src/ig/IGExporter.ts

```diff
@@ -962,8 +966,39 @@ export class IGExporter {
           }
           const resourceJSON: InstanceDefinition = this.fhirDefs.getPredefinedResource(file);
           if (resourceJSON) {
+            // For predefined examples of Logical Models, the user must provide an entry in config
+            // that specifies the reference as Binary/[id], the extension that specifies the resource format,
+            // and the exampleCanonical that references the LogicalModel the resource is an example of.
+            // In that case, we do not want to add our own entry for the predefined resource - we just
+            // want to use the resource entry from the sushi-config.yaml
+            const hasBinaryExampleReference = (this.config.resources ?? []).some(
+              resource =>
+                resource.reference?.reference === `Binary/${resourceJSON.id}` &&
+                resource.exampleCanonical ===
+                  `${this.config.canonical}/StructureDefinition/${resourceJSON.resourceType}` &&
+                resource.extension?.some(e => igResourceFormatExtensionUrls.includes(e.url))
+            );
+
+            // This implementation is based on discussion on Zulip here:
+            // https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/How.20do.20I.20get.20SUSHI.20to.20ignore.20a.20binary.20JSON.20logical.20instance.3F/near/407861211
+            const hasBinaryResourceReference = (this.config.resources ?? []).some(
+              resource =>
+                resource.reference?.reference === `Binary/${path.parse(file).name}` &&
+                resource.extension?.some(e => igResourceFormatExtensionUrls.includes(e.url))
+            );
```

Comment:
I think we should consider optimizing this a bit. Right now, for every predefined resource, it will loop through the sushi-config resource list twice.

We could combine the `hasBinaryExampleReference` and `hasBinaryResourceReference` into one loop that checks for the id-based Binary reference plus exampleCanonical, OR the file-based Binary reference. That would get us down to one loop per predefined resource.

OR... we could create a `configuredBinaryReferences` variable _before_ the top-level for loop (line 955-956) initialized using `(this.config.resources ?? []).filter(...)` to get the list of resource references starting with `Binary/` and having one of those extensions on it. Then just check against the `configuredBinaryReferences` list of resources here.

Reply:
I like both of these simplifications, but I think I'm leaning towards the second one. I'll try it out and pick one!

Reply:
Went with the second one in [047b02b](https://github.com/FHIR/sushi/pull/1418/commits/047b02bd6bdd1df628029f3ab40a8939272b22b4).